libcxx/include/__format/unicode.h

```diff
@@ -292,84 +294,231 @@ class __code_point_view<wchar_t> {
 };
 #    endif // _LIBCPP_HAS_NO_WIDE_CHARACTERS
 
-_LIBCPP_HIDE_FROM_ABI constexpr bool __at_extended_grapheme_cluster_break(
-    bool& __ri_break_allowed,
-    bool __has_extened_pictographic,
-    __extended_grapheme_custer_property_boundary::__property __prev,
-    __extended_grapheme_custer_property_boundary::__property __next) {
-  using __extended_grapheme_custer_property_boundary::__property;
+// State machine to implement the Extended Grapheme Cluster Boundary
```

Comment:
Do we not need any updated tests for these changes to the algorithm?

Reply:
Yes we do need updates, and I did. See my comment below.